### PR TITLE
#5153 Add custom cy commands for creating and connecting Paper elements

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist

- [x] [Review Contribution Guidelines](https://oharastream.github.io/en/docs/master/contributing/)
- [x] Add at least one reviewer (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add an assignee (of course, you are the assignee by default)
- [x] Do not set milestone, projects and labels (PR labels are added automatically by Jenkins)
- [x] Link to an issue
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (If the failed tests are unrelated to this PR, add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/it/paper/toolbox.test.ts [hunk 1]
```diff
@@ -214,41 +214,37 @@ describe('ToolBox', () => {
...
-        type,
+        className,
       }));
 
-      const sinks = Object.values(SINKS).map((type) => ({
+      const sinks = Object.values(SINKS).map((className) => ({
         name: generate.serviceName({ prefix: 'sink' }),
         kind: KIND.sink,
-        type,
+        className,
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/paper/toolbox.test.ts:226]: Wrong argument name...

File: ohara-manager/client/cypress/support/customCommands.ts [hunk 2]
```diff
@@ -43,6 +44,12 @@ interface FixtureResponse {
   tags?: object;
 }
 
+export interface ElementParameters {
+  name: string;
+  kind: string;
```
Comment (konekoya) [inline ohara-manager/client/cypress/support/customCommands.ts:49]: @saivirtue Do you have any idea of how to constraint the "kind" to only accept a range of string?

I tried to do:
```js
kind: KIND.stream | KIND.topic | KIND.source | KIND.sink | KIND.shabondi
```

And used it like:
```js
const elements = [
  {
    name: generate.serviceName({ prefix: 'source' }),
    kind: KIND.source,
    className: SOURCES.jdbc,
  },
  {
    name: generate.serviceName({ prefix: 'sink' }),
    kind: KIND.sink,
    className: SINKS.hdfs,
  },
  {
    name: sharedTopicName,
    kind: KIND.topic,
  },
  {
    name: 'T1',
    kind: KIND.topic,
  },
  {
    name: generate.serviceName({ prefix: 'stream' }),
    kind: KIND.stream,
  },
];

cy.addElements(elements);
```

But TS is not very happy about this and giving me:
```
 Types of property 'kind' are incompatible.
        Type 'KIND' is not assignable to type 'KIND.stream | KIND.sink | KIND.source | KIND.shabondi | KIND.topic'
```

Comment (saivirtue) [reply ohara-manager/client/cypress/support/customCommands.ts:48]: IIRC, you cannot combine `enum` with `literal types` 🤔 

Still, there are some workarounds...
https://stackoverflow.com/questions/52393730/typescript-string-literal-union-type-from-enum

Comment (konekoya) [reply ohara-manager/client/cypress/support/customCommands.ts:48]: Thanks, I will stick to "string" then as the workaround is not that pretty ☹️

Comment (konekoya) [reply ohara-manager/client/cypress/support/customCommands.ts]: Thanks for the help @saivirtue ❤️

File: ohara-manager/client/cypress/support/pipelineCommands.ts [hunk 3]
```diff
@@ -239,16 +229,53 @@ Cypress.Commands.add('addElement', (name, kind, className) => {
     }
 
     // wait for the cell added
-    cy.get('#outline').within(() => {
-      cy.findByText(name).should('exist');
+    cy.get('#outline').findByText(name).should('exist');
+
+    // If it's a topic, we need to make sure it's started and in 'running' state
+    // this is because 'pending' topic cannot be operated
+    cy.get('#paper').then(($paper) => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/support/pipelineCommands.ts:236]: This should fix some of our flakey tests that are trying to create the connection before a topic is ready (running, not in pending state)

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
Fail test is not related to this PR

**Pull request merged**